contracts/src/v0.8/llo-feeds/v0.5.0/Verifier.sol

```diff
@@ -0,0 +1,346 @@
+      bytes32 rawVs
+    ) = abi.decode(signedReport, (bytes32[3], bytes, bytes32[], bytes32[], bytes32));
+
+    // reportContext consists of:
+    // reportContext[0]: ConfigDigest
+    // reportContext[1]: 27 byte padding, 4-byte epoch and 1-byte round
+    // reportContext[2]: ExtraHash
+    bytes32 configDigest = reportContext[0];
+
+    VerifierState storage verifierState = s_verifierStates[configDigest];
+
+    _validateReport(configDigest, rs, ss, verifierState);
+
+    bytes32 hashedReport = keccak256(reportData);
+
+    _verifySignatures(hashedReport, reportContext, rs, ss, rawVs, verifierState);
+    emit ReportVerified(bytes32(reportData), sender);
+
+    return reportData;
+  }
+
+  /// @notice Validates parameters of the report
+  /// @param configDigest Config digest from the report
+  /// @param rs R components from the report
+  /// @param ss S components from the report
+  /// @param config Config for the given digest
+  function _validateReport(
+    bytes32 configDigest,
+    bytes32[] memory rs,
+    bytes32[] memory ss,
+    VerifierState storage config
+  ) private view {
+    uint8 expectedNumSignatures = config.f + 1;
+
+    if (!config.isActive) revert DigestInactive(configDigest);
+    if (rs.length != expectedNumSignatures) revert IncorrectSignatureCount(rs.length, expectedNumSignatures);
+    if (rs.length != ss.length) revert MismatchedSignatures(rs.length, ss.length);
+  }
+
+  /// @notice Verifies that a report has been signed by the correct
+  /// signers and that enough signers have signed the reports.
+  /// @param hashedReport The keccak256 hash of the raw report's bytes
+  /// @param reportContext The context the report was signed in
+  /// @param rs ith element is the R components of the ith signature on report. Must have at most MAX_NUM_ORACLES entries
+  /// @param ss ith element is the S components of the ith signature on report. Must have at most MAX_NUM_ORACLES entries
+  /// @param rawVs ith element is the the V component of the ith signature
+  /// @param config The config digest the report was signed for
+  function _verifySignatures(
+    bytes32 hashedReport,
+    bytes32[3] memory reportContext,
+    bytes32[] memory rs,
+    bytes32[] memory ss,
+    bytes32 rawVs,
+    VerifierState storage config
+  ) private view {
+    bytes32 h = keccak256(abi.encodePacked(hashedReport, reportContext));
+    // i-th byte counts number of sigs made by i-th signer
+    uint256 signedCount;
+
+    Signer memory o;
+    address signerAddress;
+    uint256 numSigners = rs.length;
+    for (uint256 i; i < numSigners; ++i) {
+      signerAddress = ecrecover(h, uint8(rawVs[i]) + 27, rs[i], ss[i]);
+      o = config.oracles[signerAddress];
+      if (o.role != Role.Signer) revert BadVerification();
+      unchecked {
+        signedCount += 1 << (8 * o.index);
+      }
+    }
+
+    if (signedCount & ORACLE_MASK != signedCount) revert BadVerification();
+  }
+
+  /// @inheritdoc IVerifier
+  function updateConfig(
```

Comment:
This looks quite similar to an OCR interface. Any reason not to extend off of an OCR contract (e.g. OCR2 - https://github.com/smartcontractkit/chainlink/tree/0b9bcaeb2bb5c9c95cc2a9d2859f2c04f25a72fc/contracts/src/v0.8/shared/ocr2) / rename the `Verifier` against OCR terms?


E.g. CCIP has their own custom version of the OCR implementation, but they still base all documentation & terms on OCR - https://github.com/smartcontractkit/ccip/blob/ccip-develop/contracts/src/v0.8/ccip/ocr/MultiOCR3Base.sol

Reply:
The interface is slightly different although the contracts were loosely based of OCR contracts. However in the next iteration we do plan on considering moving back to the setConfig model where it deterministically generates the digest. 

But this contract is no longer associated with the OCR protocol configuration, that is now the Configurator contracts job. This contract is now responsible for processing reports based on the configurations that have been set by ourselves